html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm

```diff
@@ -209,7 +209,10 @@ sub execute_child {
     # HACK alert : E-mail registration has the user registered but still going in the portal
     # release_bypass is there for that. If it is set, it will keep the user in the portal
     my $node = node_view($self->current_mac);
-    if(defined($node->{status}) && $node->{status} eq "reg" && !$self->app->session->{release_bypass}){
+    if($self->app->profile->canAccessRegistrationWhenRegistered() && $self->app->session->{release_bypass}) {
+        get_logger->info("Allowing user through portal even though he is registered as the release bypass is set and the portal profile is configured to let registered users use the registration module of the portal.");
+    }
+    elsif(defined($node->{status}) && $node->{status} eq "reg"){
```

Comment:
No, that isn't necessary because unless $self->app->profile->canAccessRegistrationWhenRegistered() is true, we don't want to consider release_bypass